# Praca domowa 6

## Zadanie 1


In [35]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv("house_data.csv")

In [10]:
pd.set_option('display.max_columns', None)
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  condition  grade  sqft_above  \
0      5650     1.0           0     0          3      7        1180   
1      7242     2.0           0     0          3      7        2170   
2     10000     1.0           0     0          3      6         770   
3      5000     1.0           0     0          5      7        1050   
4      8080     1.0           0     0          3      8        1680   

   sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0              0      1955             0    98178  47.5112 -122.257   
1            400      1951          1991    98125  47.7210 -122.319   
2              0      1933             0    98028  47.7379 -122.233   
3            910      1965             0    98136  47.5208 -122.393   
4              0      1987             0    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503

1. Jaka jest średnia cena nieruchomości położonych nad wodą, których jakoś wykończenia jest równa lub większa od mediany jakości wykończenia?


In [14]:
df.price.loc[(df.waterfront == 1) & (df.grade >= df.grade.median())].mean()

1784151.7006802722

Odp: 1784151.7006802722

2. Czy nieruchomości o 2 piętrach mają większą (w oparciu o wartości mediany) liczbę łazienek niż nieruchomości o 3 piętrach?


In [15]:
df.bathrooms.loc[df.floors == 2.0].median() > df.bathrooms.loc[df.floors == 3.0].median()

False

Odp: nie

3. Jak wygląda wartość kwartyla 0.25 oraz 0.75 jakości wykończenia nieruchomości położonych na północy biorąc pod uwagę czy ma ona widok na wodę czy nie ma?

In [45]:
df.loc[df.lat > ((df.lat.max()+df.lat.min()) / 2)].groupby("waterfront").grade.describe()

count      mean       std  min  25%  50%   75%   max
waterfront                                                        
0           16198.0  7.719965  1.217724  1.0  7.0  8.0   8.0  13.0
1             115.0  9.139130  1.695644  5.0  8.0  9.0  10.0  12.0

Odp: z widokiem: q1 = 8, q3 = 10
     bez widoku: q1 = 7, q3 = 8

4. Pod którym kodem pocztowy jest położonych najwięcej nieruchomości i jaki jest rozstęp miedzykwartylowy dla ceny nieruchomości położonych pod tym adresem?

In [31]:
df.groupby("zipcode").id.count().sort_values(ascending = False).head(1)
q3 = np.percentile(df.loc[df.zipcode == 98103, "price"], 75)
q1 = np.percentile(df.loc[df.zipcode == 98103, "price"], 25)
q3-q1

262875.0

Odp: pod kodem 98103, i rozstep wynosi 262875

5. Ile procent nieruchomości ma wyższą średnią powierzchnię 15 najbliższych sąsiadów względem swojej powierzchni?

In [37]:
df.loc[df.sqft_lot15 > df.sqft_lot].id.count() / df.id.count() * 100

39.513255910794435

Odp: około 39,5% dla powierzchni działki

 6. O ile procent więcej jest nieruchomości leżcych na północy zachód niż  nieruchomości leżących na południowy wschód?

In [46]:
numNW = df.loc[(df.lat > ((df.lat.max() + df.lat.min()) / 2)) & (df.long < (df.long.max() + df.long.min()) / 2)].id.count() 
numSE = df.loc[(df.lat < ((df.lat.max() + df.lat.min()) / 2)) & (df.long > (df.long.max() + df.long.min()) / 2)].id.count()
abs(numNW - numSE) / numSE * 100

18343.529411764706

Odp: o około 18343%

# Zadanie 2

Dla wybranej zmiennej ze zbioru danych z Zadania 1 narysuj:

a) rozkład zmiennej jakościowej

b) rozkład zmiennej ilościowej

c) rozkład dwóch zmiennych (dowolna kombinacja jakościowej i ilościowej)

In [47]:
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  condition  grade  \
0             1180      5650     1.0           0     0          3      7   
1             2570      7242     2.0           0     0          3      7   
2              770     10000     1.0           0     0          3      6   
3             1960      5000     1.0           0     0          5      7   
4             1680      8080     1.0           0     0          3      8   
...            ...       ...     ...         ...   ...        ...    ...   
21608         1530      1131     3.0           0     0          3      8   
21609         2310      5813     2.0           0     0          3      8   
21610         1020      1350     2.0           0     0          3      7   
21611         1600      2388     2.0           0     0          3      8   
21612         1020      1076     2.0           0     0          3      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [77]:
import matplotlib.pyplot as plt
import seaborn as sns

a)

In [75]:
plt.hist(df.grade, bins = 9)
plt.title("Jakość wykończenia")
plt.xlabel("Stopień wykończenia")
plt.ylabel("Liczba nieruchomości")
plt.show()

C:\Users\stani\AppData\Local\Temp\ipykernel_4920\395607996.py:5: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
  plt.show()


b)

In [78]:
sns.boxplot(df.price, color='w')
plt.show()

c:\Users\stani\anaconda3\lib\site-packages\seaborn\_decorators.py:36: FutureWarning: Pass the following variable as a keyword arg: x. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(
C:\Users\stani\AppData\Local\Temp\ipykernel_4920\2978355305.py:2: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
  plt.show()


c)

In [80]:
floors = ["1", "2", "3"]
values = [df.loc[df.floors == 1].price.mean(), df.loc[df.floors == 2].price.mean(), df.loc[df.floors == 3].price.mean()]
plt.bar(floors, values)
plt.xlabel("Liczba pięter")
plt.ylabel("Średnia cena")
plt.show()

C:\Users\stani\AppData\Local\Temp\ipykernel_4920\3858027654.py:6: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
  plt.show()
